launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -455,6 +542,36 @@ private ProcessBuilder createBuilder() {
     for (Map.Entry<String, String> e : builder.childEnv.entrySet()) {
       pb.environment().put(e.getKey(), e.getValue());
     }
+
+    if (builder.workingDir != null) {
+      pb.directory(builder.workingDir);
+    }
+
+    // Only one of redirectError, redirectError(...) can be specified.
+    // If redirectToLog() is specified, none of the other redirect options can be specified
+    if (builder.redirectToLog &&
+            (builder.redirectErrorStream
+                    || (builder.errorStream != null)
+                    || (builder.outputStream != null))) {
+      throw new IllegalStateException("Cannot specify redirectToLog() in conjunction with other " +
+              "redirection operations.");
+    }
+
+    // Only one of redirectError and redirectError(...) can be specified
+    if (builder.redirectErrorStream && builder.errorStream != null) {
```

Comment:
Probably ok to leave this one, but `ProcessBuilder` allows it even if it doesn't do what you'd expect.

```
new ProcessBuilder(java.util.Arrays.asList("ls", "/foo")).redirectErrorStream(true).redirectError(ProcessBuilder.Redirect.INHERIT).start()
```